### DISCUSSION TITLE
Content Security Policy Error while fetching Audio files from Directus Hosted Project

### DISCUSSION THREAD

**Discussion by Leotheluck** [directus/directus#13417] [Archive]
Hello everyone, 

I'm using Directus Cloud to host audio files which I retrieve in my application and play. Everything did work fine but now when I'm fetching the audio file it is no longer playing. When I try the url online I get this error:

`Refused to load media from 'https://myapp.directus.app/assets/audiofileid' because it violates the following Content Security Policy directive: "default-src none". Note that 'media-src' was not explicitly set, so 'default-src' is used as a fallback.`

I have seen some tweaks with the CSP but I have no idea where to access these parameters since I am using Directus Cloud and have no other server or anything to edit. 

Other files such as images do work though, so I don't really understand how this error works and how I could fix it, so help would be very much appreciated! Thanks! ☀️

**Comment by azrikahar** [comment]
cc @joselcvarela

**Reply by joselcvarela** [reply]
This seems to be an issue within Directus itself rather than Cloud.
I mean, it's possible to configure environment variables, but by default it's not possible to directly access media assets like videos or audios.
That's because here we are restricting it to  `default-src: none`.
https://github.com/directus/directus/commit/a8df6d96531fea4ee747594d0b047d2a676714be

@rijkvanzanten (because you introduced that change) any reason we do not use `default-src: 'self'` (quotes on purpose) ?
It seems to be as safe as before and they can be reproduced like in a new tab.
For example:
`default-src: none` and accessing `http://localhost:8055/assets/3b84f6c4-8ce9-4192-9907-12f28b76a143` on new tab does not work.
But
`default-src: 'self'` and accessing `http://localhost:8055/assets/3b84f6c4-8ce9-4192-9907-12f28b76a143` on new tab works.

**Reply by rijkvanzanten** [reply]
~~The last two examples are the same it seems.~~

The current configuration is based on the "least privilege", so it's probably a little too tight for some uses

**Reply by joselcvarela** [reply]
Updated: should be `default-src: 'self'` instead of `default-src: none`

**Comment by rijkvanzanten** [comment]
> Refused to load media from 'https://myapp.directus.app/assets/audiofileid' because it violates the following Content Security Policy directive: "default-src none". Note that 'media-src' was not explicitly set, so 'default-src' is used as a fallback.

This is actually correct, to prevent malicious files that may be uploaded from your users from executing in the browser, however, that shouldn't prevent the file to be loaded from an external page in an <audio> tag. How are you currently accessing the audio file in your project?

**Reply by Leotheluck** [reply]
Currently I'm accessing the file like this (passing the uri through props)
Loading it as soon as the screen is displayed (useEffect)

`const { sound: playbackObject } = await Audio.Sound.createAsync({uri: props.link})`
`sound.current = playbackObject`

**Reply by rijkvanzanten** [reply]
And is that throwing any errors? I know that opening the audio file directly in the browser will throw a warning, but does that particular snippet throw anything?

**Reply by Leotheluck** [reply]
It was until I retried just now 😭 We had an alternative fix with firebase hosting for the files and now I tried re-implementing it the old way and it did work this time, maybe the error was elsewhere? or did I just need to reboot my computer in between, or something else?

Anyways, thanks for your help, I'll keep it updated here if I do get the bug again or find the reason why it was blocked 🤞

**Reply by rijkvanzanten** [reply]
Who knows! Such are the wonders of modern web development 🚀 

Glad to hear it seems to be working as it should now though 🙂

**Comment by arladmin** [comment]
Hi @rijkvanzanten, I'm facing the same issue when trying to access a media (video) file using the Assets endpoint, via the browser.
```
Refused to load media from 'https://mym-domain/assets/id' because it violates the following Content Security Policy directive: "default-src none". Note that 'media-src' was not explicitly set, so 'default-src' is used as a fallback.
```
What can i do to make sure that this link resolves successfully in the browser?

**Reply by azrikahar** [reply]
Can you try setting the `ASSETS_CONTENT_SECURITY_POLICY__MEDIA_SRC` environment variable? The exact value will differ depending on what CSP error you will get, but perhaps you can try `ASSETS_CONTENT_SECURITY_POLICY__MEDIA_SRC="array:'self',your-domain.com"` as a starting point (`your-domain.com` is just an example, make sure to change to the one in your error).

You can also refer to [this comment](https://github.com/directus/directus/discussions/15865#discussioncomment-3786705) in case you are having trouble "translating" [helmet.js `contentSecurityPolicy` options](https://github.com/helmetjs/helmet#reference) into environment variables in Directus.

**Comment by Posoroko** [comment]
`but perhaps you can try ASSETS_CONTENT_SECURITY_POLICY__MEDIA_SRC="array:'self',your-domain.com" as a starting point `

I have the same problem when trying to read videao files from the assets endpoint. 
I tried a bunch of different values for this env variable, but nothing helps.
But, I run Directus as a Docker image through Plesk.  I usually have no problem setting up anv variable, but I'm not sure how to handle the "array:'self'" syntax.  
Does anyone have more information about that problem?

**Comment by shasha79** [comment]
I has the same problem, this comment worked for me:
https://github.com/directus/directus/issues/12493#issuecomment-1084229767
